toreview/KadActionsBuilder.java

```diff
@@ -0,0 +1,311 @@
+package com.gruppo1.distributednetworkmanager;
+
+import com.dezen.riccardo.networkmanager.StringResource;
+import com.dezen.riccardo.smshandler.SMSPeer;
+
+import java.util.Random;
+
+/**
+ * @author Niccolo' Turcato
+ * Builds KadAction objs
+ * <p>
+ * Based on:
+ * https://github.com/turcato-niccolo/SmsHandler
+ */
+public class KadActionsBuilder implements ActionsBuilder<KadAction, SMSPeer, StringResource, Node<BinarySet>> {
+
+
+    private static int MAX_ACTION_ID;
+    private static int DEFAULT_PARTS = KadAction.MIN_PARTS;
+    private static final String PAYLOAD_IGNORED = "0w0";
+    private static final String ILLEGAL_MAX_ID_MSG = "Can't initialize Builder, the defined max ID isn't compatible with class KadActions";
+    public static final String RESOURCE_SEPARATOR = KadAction.RESOURCE_SEPARATOR;
+
+    public static final KadAction INVALID_ACTION = KadAction.INVALID_KAD_ACTION;

```

Comment:
importa quella costante, invece di ridefinirla. Non esistono mai due costanti identiche.

Reply:
Hai ragione, avrei dovuto fare così, il fatto è che la "funzionalità" della azione invalida è stata implementata durante lo sviluppo di questa classe e avevo lasciato temporaneamente questa costante a cui poi ho dato il valore KadAction.INVALID_KAD_ACTION per testarla in maniera pratica.
Farò questo refactoring quanto prima